run.go

```diff
@@ -0,0 +1,134 @@
+package autopilot
+
+import (
+	"context"
+	"time"
+)
+
+// Start will launch the go routines in the background to perform Autopilot.
+// When the context passed in is cancelled or the Stop method is called
+// then these routines will exit.
+func (a *Autopilot) Start(ctx context.Context) {
+	a.runLock.Lock()
+	defer a.runLock.Unlock()
+
+	// already running so there is nothing to do
+	if a.running {
+		return
+	}
+
+	ctx, shutdown := context.WithCancel(ctx)
+	a.shutdown = shutdown
+	a.startTime = a.time.Now()
+	a.done = make(chan struct{})
+
+	// While a go routine executed by a.run below will periodically
+	// update the state, we want to go ahead and force updating it now
+	// so that during a leadership transfer we don't report an empty
+	// autopilot state. We put a pretty small timeout on this though
+	// so as to prevent leader establishment from taking too long
+	updateCtx, updateCancel := context.WithTimeout(ctx, time.Second)
+	defer updateCancel()
+	a.updateState(updateCtx)
```

Comment:
We don't capture the returned error from the `updateState` function in both usages of it. I can see an argument for ignoring it, since we want to continue processing and the function itself already logs an error. In that case though, we should remove the `error` return value altogether.

Reply:
Removed the return